onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -5609,13 +5613,14 @@ def _aten_native_batch_norm_training_onnx(
         training_mode=training,
     )
     # Compute var and rstd
-    mean = op.ReduceMean(input, axes)
-    input_sub_mean = op.Sub(input, mean)
+    upcast_input = op.Cast(input, to=FLOAT.dtype)
+    mean = op.ReduceMean(upcast_input, axes)
+    input_sub_mean = op.Sub(upcast_input, mean)
     sqr = op.Mul(input_sub_mean, input_sub_mean)
     var = op.ReduceMean(sqr, axes, keepdims=False)
     rstd = op.Div(1.0, op.Sqrt(var + eps))
     # Get mean again with size = [1, C]
-    mean = op.ReduceMean(input, axes, keepdims=False)
+    mean = op.ReduceMean(upcast_input, axes, keepdims=False)
     return norm, mean, rstd
 
```

Comment:
Do we need to tighten the test tolerance if there is any set?

Reply:
idk, I have no idea which test case is associated with this particular scenario ...

Reply:
It is specified here: https://github.com/microsoft/onnxscript/blob/7428f920ebb9713ec78bae7555090f0f55e05ef0/onnxscript/tests/function_libs/torch_lib/ops_test_data.py#L1813

Reply:
But may be unrelated if training is never true in the test cases

Reply:
Feel free to merge if removing this line fails the tests